### PR TITLE
[hackathon7] Add timit/asr1 readme.md

### PR DESCRIPTION
**Pull request by enkilee**
### PR types

Others
### PR changes

Docs
### Describe

Add readme file for timit's asr1

### CODE REVIEW

File: examples/timit/asr1/README.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
+# Transformer/Conformer ASR with Timit
+The phoneme-based continuous speech corpus is a collaboration between Texas Instruments, MIT, and SRI International. The [Timit](https://catalog.ldc.upenn.edu/docs/LDC93S1/) dataset has a voice sampling frequency of 16 khz and contains a total of 6,300 sentences, with 630 people from eight major U.S. dialects speaking a given 10 sentences each, all sentences are manually segmented and marked at the phone level. Seventy percent of the speakers are male; most of the speakers are white adults.
+
+## Dataset
+### Download and Extract
+Download TIMIT from it's [official website](https://catalog.ldc.upenn.edu/LDC93S1) and extract it to `~/datasets`. Then the dataset is in the directory `~/datasets/timit`.
```
Comment (zxcd) [inline examples/timit/asr1/README.md]: data directory should same with `run.sh`.

File: examples/timit/asr1/README.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+`stop stage` denotes the number of the stage you want to end at in the experiments. 
+`conf_path` denotes the config path of the model.
+`avg_num` denotes the number K of top-K models you want to average to get the final model.
+`audio_file` denotes the file path of the single file you want to infer in stage 5
+`ckpt` denotes the checkpoint prefix of the model, e.g. "conformer"
+You can set the local variables (except `ckpt`) when you use `run.sh`
+
+For example, you can set the `gpus` and `avg_num` when you use the command line.:
+```bash
+bash run.sh --gpus 0,1 --avg_num 20
```
Comment (zxcd) [inline examples/timit/asr1/README.md]: command should same with `run.sh`

File: examples/timit/asr1/README.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+Stage 0 is for processing the data.
+
+If you only want to process the data. You can run
+```bash
+bash run.sh --stage 0 --stop_stage 0
+```
+You can also just run these scripts in your command line.
+```bash
+source path.sh
+bash ./local/timit_data_prep.sh /path/to/TIMIT
```
Comment (zxcd) [inline examples/timit/asr1/README.md]: this path should be fixed by user

File: examples/timit/asr1/README.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+ fi
+```
+If you want to train the model, you can use the script below to execute stage 0 and stage 1:
+```bash
+bash run.sh --stage 0 --stop_stage 1
+```
+or you can run these scripts in the command line (only use CPU).
+```bash
+source path.sh
+bash ./local/timit_data_prep.sh /path/to/TIMIT
```
Comment (zxcd) [inline examples/timit/asr1/README.md]: same with above

File: examples/timit/asr1/README.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+bash ./local/data.sh
+CUDA_VISIBLE_DEVICES= ./local/train.sh conf/transformer.yaml transformer
```
Comment (zxcd) [inline examples/timit/asr1/README.md]: why don't use gpu

File: examples/timit/asr1/README.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+```bash
+bash run.sh --stage 0 --stop_stage 2
+```
+or you can run these scripts in the command line (only use CPU).
+```bash
+bash ./local/timit_data_prep.sh /path/to/TIMIT
+source path.sh
+bash ./local/data.sh
+CUDA_VISIBLE_DEVICES= ./local/train.sh conf/transformer.yaml transformer
+avg.sh best exp/conformer/checkpoints 20
```
Comment (zxcd) [inline examples/timit/asr1/README.md]: same with above

File: examples/timit/asr1/README.md [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+```
+## Stage 3: Model Testing
+The test stage is to evaluate the model performance. The code of the test stage is shown below:
+```bash
+ if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ]; then
+     # test ckpt avg_n
+     CUDA_VISIBLE_DEVICES=0 ./local/test.sh ${conf_path} exp/${ckpt}/checkpoints/${avg_ckpt} || exit -1
+ fi
+```
+If you want to train a model and test it, you can use the script below to execute stage 0, stage 1, stage 2, and stage 3 :
```
Comment (zxcd) [inline examples/timit/asr1/README.md:155]: attention the spaces at the end of sentence

File: examples/timit/asr1/README.md [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+```bash
+bash run.sh --stage 0 --stop_stage 3
+```
+or you can run these scripts in the command line (only use CPU).
+```bash
+source path.sh
+bash ./local/timit_data_prep.sh /path/to/TIMIT
+bash ./local/data.sh
+CUDA_VISIBLE_DEVICES= ./local/train.sh conf/transformer.yaml transformer
+avg.sh best exp/transformer/checkpoints 20
```
Comment (zxcd) [inline examples/timit/asr1/README.md]: same with above

File: examples/timit/asr1/README.md [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+CUDA_VISIBLE_DEVICES= ./local/test.sh conf/transformer.yaml exp/transformer/checkpoints/avg_20
```
Comment (zxcd) [inline examples/timit/asr1/README.md]: same with above

File: examples/timit/asr1/README.md (hunk 1 above)
Comment (enkilee) [reply examples/timit/asr1/README.md:6]: I've not unzip the TIMIT.zip to the same path with run.sh.
I've unzip the TIMIT.zip to the datasets.

File: examples/timit/asr1/README.md [hunk 10]
```diff
@@ -0,0 +1,195 @@
+# Transformer/Conformer ASR with Timit
```
Comment (zxcd) [inline examples/timit/asr1/README.md]: seems without conformer training in this config. Also fix `examples/timit/README.md`

File: examples/timit/asr1/README.md [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+The phoneme-based continuous speech corpus is a collaboration between Texas Instruments, MIT, and SRI International. The [Timit](https://catalog.ldc.upenn.edu/docs/LDC93S1/) dataset has a voice sampling frequency of 16 khz and contains a total of 6,300 sentences, with 630 people from eight major U.S. dialects speaking a given 10 sentences each, all sentences are manually segmented and marked at the phone level. Seventy percent of the speakers are male; most of the speakers are white adults.
```
Comment (zxcd) [inline examples/timit/asr1/README.md]: Correct the grammar of this sentence, and unify the usage of numbers.
Such `6,300 sentences, with 630 people from eight major` should be fixed like `6,300 sentences, with 630 speakers from 8 major dialect regions of U.S.`.

File: examples/timit/asr1/README.md (hunk 10 above)
Comment (enkilee) [reply examples/timit/asr1/README.md]: roger

File: examples/timit/asr1/README.md (hunk 11 above)
Comment (enkilee) [reply examples/timit/asr1/README.md]: roger

**Review by zxcd — APPROVED** [review]
LGTM

**Pull request merged**